src/components/user/CreateTargetForm.js

```diff
@@ -92,31 +99,46 @@ export class CreateTargetForm extends PureComponent {
             options={topics}
             type="select"
             placeholder={intl.formatMessage(messages.placeHolderTopics)}
+            selectedOption={topicIdSelected}
+            disabled={isDeletingTarget}
           />
         </div>
         {error && <strong>{error}</strong>}
         <div className="wrapper-button">
+          {isDeletingTarget &&
+          <button className="delete-button" type="submit" >
+            <FormattedMessage id="target.form.deleteTarget" />
+          </button>
+          }
           <button className="create-target__button" type="submit">
             <FormattedMessage id="target.form.saveTarget" />
           </button>
         </div>
 
+        {submitting && <Loading />}
+        {submitSucceeded && <div className="icon saved-icon--small" />}
+
         <div>
           <img id="smiles-icon" className="icon smile-icon--small" alt="smiles" src={smileIcon} />
         </div>
-        {submitting && <Loading />}
       </form>
     );
   }
 }
 
 CreateTargetForm = reduxForm({
   form: 'target',
-  validate: validations(createTarget, { fullMessages: false })
+  validate: validations(createTarget, { fullMessages: false }),
+  onSubmitSuccess: (result, dispatch, props) => {
```

Comment:
I'm curious, why the need for this?

Reply:
I need to reset the form after the submit when it´s successful.

Reply:
I need to reset the form after submit when it´s successful.